sbncode/BeamSpillInfoRetriever/getFOM.cpp

```diff
@@ -0,0 +1,410 @@
+       interpolate_vp875(vptg1[0] - vptg1_offset, vptg1_zpos):
+       interpolate_vp875(vptg2[0] - vptg2_offset, vptg2_zpos);
+
+    horang=atan(horang);
+    verang=atan(verang);
+    double xx,yy,sx,sy,chi2x,chi2y;
+    double tgtsx, tgtsy;
+    bool good_tgt=false;
+    bool good_876=false;
+    bool good_875=false;
+	constexpr size_t FirstXMWtgt =0;
+	constexpr size_t FirstYMWtgt =48;
+	const double smallSigmaX =0.5, largeSigmaX = 10, smallSigmaY = 0.3, largeSigmaY =10, maxChi2X = 20, maxChi2Y = 20;
+    if (mwtgt.size()>0) {
+      processBNBprofile(&mwtgt[FirstXMWtgt], xx, sx,chi2x);
+      processBNBprofile(&mwtgt[FirstYMWtgt], yy, sy, chi2y);
+      if (sx>smallSigmaX && sx<largeSigmaX && sy>smallSigmaY && sy<largeSigmaY && chi2x<maxChi2X && chi2y<maxChi2Y) {
+		tgtsx=sx;
+		tgtsy=sy;
+		good_tgt=true;
+      }
+    }
+    if (!good_tgt && mw876.size()>0) {
+      processBNBprofile(&mw876[FirstXMWtgt], xx,sx,chi2x);
+      processBNBprofile(&mw876[FirstYMWtgt], yy,sy,chi2y);
+      double tgtsx876=p876x[0]+p876x[1]*sx+p876x[2]*sx*sx;
+      double tgtsy876=p876y[0]+p876y[1]*sy+p876y[2]*sy*sy;
+      if (tgtsx876>smallSigmaX && tgtsx876<largeSigmaX && tgtsy876>smallSigmaY && tgtsy876<largeSigmaY && chi2x< maxChi2X && chi2y< maxChi2Y) {
+		tgtsx=tgtsx876;
+		tgtsy=tgtsy876;
+		good_876=true;
+      }
+    }
+    if (!good_tgt && !good_876 && mw875.size()>0){
+      processBNBprofile(&mw875[FirstXMWtgt], xx,sx,chi2x);
+      processBNBprofile(&mw875[FirstYMWtgt], yy,sy,chi2y);
+      double tgtsx875=p875x[0]+p875x[1]*sx+p875x[2]*sx*sx;
+      double tgtsy875=p875y[0]+p875y[1]*sy+p875y[2]*sy*sy;
+      if (tgtsx875>smallSigmaX && tgtsx875<largeSigmaX && tgtsy875>smallSigmaY && tgtsy875<largeSigmaY && chi2x<maxChi2X && chi2y<maxChi2Y) {
+		tgtsx=tgtsx875;
+		tgtsy=tgtsy875;
+		good_875=true;
+      }
+    }
+    if (!good_tgt && !good_876 && !good_875) {
+      //failed getting  multiwire data
+      return 4;
+    }
+    
+    fom=1-pow(10,sbn::calcFOM(horpos,horang,verpos,verang,tor,tgtsx,tgtsy));
+    return fom;
+  }
+  
+
+/**
+    * @brief Extracts statistics from multiwire monitor data.
+    * @param mwdata pointer to multiwire data (48 channels horizontal or vertical)
+    * @param[out] x mean position from the fit [mm]
+    * @param[out] sx &sigma; from the fit [mm]
+    * @param[out] chi2 &chi;&sup2;/NDF for the Gaussian fit
+    * 
+    * This function takes multiwire data (`mwdata`),
+    * finds the min and max,
+    * finds the first and last bin where amplitude is greater than 20%,
+    * fits the peak between first and last bin with Gaussian (assuming 2% relative errors)
+    * and returns the parameters of the fit.
+    */
+    void processBNBprofile(const double* mwdata, double &x, double& sx, double& chi2)
+  {
+    // values' sign is inverted
+    double minx = std::min(-*std::max_element(mwdata, mwdata + 48), 0);
+    double maxx = std::max(-*std::min_element(mwdata, mwdata + 48), 0);
```

Comment:
If turns out that my old suggestion was generating a `std::min(double, int)`, which stops the compiler because it does not know whether convert the `int` to a `double`, or the `double` to an `int`. Let's make both `double`:
```suggestion
    double minx = std::min(-*std::max_element(mwdata, mwdata + 48), 0.0);
    double maxx = std::max(-*std::min_element(mwdata, mwdata + 48), 0.0);
```
Also, please test that this change works as it should, since it's a bit less than straightforward.